tests/iio_stresstest.c

```diff
@@ -388,18 +398,37 @@ static void *client_thread(void *data)
 	return (void *)EXIT_FAILURE;
 }
 
-static unsigned long int clamp(const char *name, unsigned long int val,
+static unsigned long int sanitize_clamp(const char *name, const char *val,
 		unsigned long int min, unsigned long int max)
 {
-	if (val > max) {
-		val = max;
-		fprintf(stderr, "Clamped %s to max %lu", name, max);
+
+	unsigned long int tmp;
+	char buf[20];
+
+	if(!val) {
+		tmp = 0;
+	} else {
+
+		/* find the length of chars of max, and make sure the string
+		 * is shorter than that
+		 */
+		iio_snprintf(buf, sizeof(buf), "%lu", max);
+		if (strnlen(val, strlen(buf) + 1) <= strlen(buf)) /* Flawfinder: ignore */
```

Comment:
umm, i feel this is where pleasing coverity starts to reduce the value of the code [in terms of readability, maintainability,  understanding the intent of the code, etc];
[ to-be-fair: i also [sometimes] fall easily into the trap of pleasing static-checkers ; ]

using a buffer for argv[idx]  + strlen()/strlnlen() doesn't seem to be more efficient than simply using strtol()
if there is no null terminator, things will go sour anyway;
i'd expect strltoul() to use strlen() or something equivalent anyway;

now, there could be some better way to please coverity and still make this readable;
but if there is a flag/comment for coverity to ignore this, it's probably not a bad idea [either];

[fwiw: the idea of parsing argv[] in this function is nice though];

Reply:
Updated with a simpler solution.